### ISSUE TITLE
testing on multiple python versions with tox

### ISSUE THREAD

**Issue by aewallin** [aewallin/allantools#41]
tox seems to be a solution for testing on multiple python versions:
https://tox.readthedocs.org/en/latest/

at least Ubuntu 16.04LTS looks like it is based on Python 2.7, and probably will stay that way until replaced in 2018.
On the other hand many people are already on python 3.

So automated tox testing + documentation on how to do it would be a welcome enhancement.

**Comment by mivade** [comment]
Tox works on Python 3 according to PyPI: https://pypi.python.org/pypi/tox  If using the system package of tox, there should be no problems running tests with tox on multiple versions of Python (that's the whole point of tox, after all).

Another option would be to use TravisCI for automated testing. I tend to use both for my own projects: tox to test things locally when making big changes, and TravisCI for catching everything else.

**Comment by aewallin** [comment]
this is fixed by pull request #42

**Comment by fmeynadier** [comment]
A comment, though : I was about to add the travis-ci adress to the README doc, but I noticed that the link from travis-ci to github is to @mivade/allantools : so it cannot be assumed to be representative of the state of @aewallin/allantools:master (unless an automated "pull --rebase" is configured somewhere ?).

I am not fully familiar with TravisCI : @mivade, what would you recommend to do ? Should another project be opened by @aewallin at travis-ci in order to link to his own repo ? or would you be able to link to it yourself ?

Anyway thanks for setting this up, this will be very useful !

**Comment by mivade** [comment]
Yes, @aewallin should enable TravisCI's automatic builds and links in the documentation should point to that in order to be pointing at the official repository. Getting it setup is pretty straightforward:

* Sign in with GitHub credentials
* Check the box for the repository

**Comment by aewallin** [comment]
This is now enabled at:
https://travis-ci.org/aewallin/allantools

**Comment by aewallin** [comment]
I added links to the 'badges' of both travis-ci and readthedocs right at the start of the readme. Hopefully that means errors or problems are spotted quickly - pretty cool!

I don't remember exactly what the dependence on scipy is (chi-squared function, and/or welch method for power spectral density?). It looks like lots of the build/test time goes to setting up scipy, so if the dependency is easy to remove it would perhaps make testing much faster.

I also noticed for example this project has three badges:
https://github.com/influxdata/influxdb-python
so there is some tool also for automatic coverage testing. I probably don't have time to look into it but if someone else wants to take a look the go ahead!
